src/CashFlowPlot.py

```diff
@@ -0,0 +1,209 @@
+    positiveCmap = plt.get_cmap('Blues')
+    positiveRawCmapList = [positiveCmap(i) for i in range(positiveCmap.N)]
+    positiveCmapList = positiveRawCmapList[defaultStartColor:defaultEndColor:(defaultEndColor-defaultStartColor)//positiveVariable]
+    cmapList = negativeCmapList + positiveCmapList # Color list
+
+    ## Plot 1: Inflows, outflows and Net Cash Flow
+    widthFlowbar = 0.35
+    _, ax1 = plt.subplots(1, figsize = (15,8))
+    for i, _ in enumerate(cumulativeCashFlow[0:negativeVariable]):
+      if i == 0:
+        ax1.bar(dfYear['cfYears'], dfCashFlow.iloc[:,i], widthFlowbar, color = cmapList[i], label = labels.iloc[i,0], edgecolor='white')
+      else:
+        ax1.bar(dfYear['cfYears'], dfCashFlow.iloc[:,i], widthFlowbar, color = cmapList[i], bottom = dfCashFlow.iloc[:,0:i-1].sum(axis=1), label = labels.iloc[i,0], edgecolor='white')
+
+    for i, _ in enumerate(cumulativeCashFlow[negativeVariable:negativeVariable+positiveVariable]):
+      if i == 0:
+        ax1.bar(dfYear['cfYears'], dfCashFlow.iloc[:,i+negativeVariable], widthFlowbar, color = cmapList[i+negativeVariable], label = labels.iloc[i+negativeVariable,0], edgecolor='white')
+      else:
+        ax1.bar(dfYear['cfYears'], dfCashFlow.iloc[:,i+negativeVariable], widthFlowbar, color = cmapList[i+negativeVariable], bottom = dfCashFlow.iloc[:,negativeVariable:negativeVariable+i].sum(axis=1), label = labels.iloc[i+negativeVariable,0], edgecolor='white')
+
+    # grid
+    ax1.set_axisbelow(True)
+    ax1.yaxis.grid(color='gray', linestyle='dashed', alpha=0.7)
+    ax1.set_xlabel('Project Year', fontsize = 15, fontweight='bold')
+    ax1.set_ylabel('Yearly Cash Flows($)', fontsize = 15, fontweight='bold')
+    # total cashflow
+    plt.plot(dfYear['cfYears'], yearNetCashFlow, color='darkolivegreen', marker='o',label='Yearly Net Cash Flow')
+    # Shrink current axis by 20% for legends
+    box = ax1.get_position()
+    ax1.set_position([box.x0, box.y0, box.width * 0.8, box.height])
+    plt.legend(loc='center left', bbox_to_anchor=(1.0,0.5), fancybox=True, ncol=1,fontsize = 15)
+    plt.title('Yearly Inflows, Outflows and Net Cash Flow from Project Year {} to {}'.format(startYear, endYear), y=1.05, fontsize = 20, fontweight='bold')
+    fName = os.path.abspath('Cash_Flows_from_Project_Year_{}_to_{}.png'.format(startYear, endYear))
+    plt.savefig(fName,bbox_inches="tight")
+    self.raiseAMessage(f'Saved figure to "{fName}"')
+
+    ## Plot 2: Cumulative discounted free cash flow
+    lableLoc = np.arange(endYear + 1 - startYear)  # the label location
+    widthCumBar = 0.1
+    _, ax1 = plt.subplots(1, figsize = (22,12))
+    for i, _ in enumerate(cumulativeCashFlow):
+      ax1.bar(lableLoc + widthCumBar * (i - (len(cumulativeCashFlow) + 1) / 2), dfCashFlow.iloc[:,i], widthCumBar, color = cmapList[i], label = labels.iloc[i,0], edgecolor='white')
```

Comment:
Some of these lines extend past the 120 line-length maximum. I think it would be valuable to reformat some of these lines for clarity in the code.